codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1764,12 +1789,15 @@ termNamesWithinNamespace bhId mayNamespace = do
           AND namespace GLOB ?
         |]
 
+-- | NOTE: requires that the codebase has an up-to-date name lookup index. As of writing, this
+-- is only true on Share.
+--
 -- | Get the list of a type names in the root namespace according to the name lookup index
 typeNamesWithinNamespace :: BranchHashId -> Maybe Text -> Transaction [NamedRef Reference.TextReference]
 typeNamesWithinNamespace bhId mayNamespace = do
   let namespaceGlob = case mayNamespace of
         Nothing -> "*"
-        Just namespace -> globEscape namespace <> ".*"
+        Just namespace -> toNamespaceGlob namespace
```

Comment:
I added helpers for this so that if we change the format of these fields again in the future there's one place to make the change.